spring-boot-starter-mesos/src/main/java/com/containersolutions/mesos/scheduler/TaskInfoFactoryDocker.java

```diff
@@ -33,39 +28,55 @@
                 .setTaskId(Protos.TaskID.newBuilder().setValue(taskId))
-                .addAllResources(resources)
+                .addAllResources(mesosResources)
                 .setContainer(Protos.ContainerInfo.newBuilder()
-                        .setType(Protos.ContainerInfo.Type.DOCKER)
-                        .setDocker(Protos.ContainerInfo.DockerInfo.newBuilder()
-                                .setImage(dockerImage)
-                                .addAllPortMappings(portMappings(resources))
-                                        .setNetwork(Protos.ContainerInfo.DockerInfo.Network.valueOf(networkMode))
-                        )
+                                .setType(Protos.ContainerInfo.Type.DOCKER)
+                                .setDocker(Protos.ContainerInfo.DockerInfo.newBuilder()
+                                                .addAllParameters(addParameters(commandInfoMesosProtoFactory.create(resources).getEnvironment()))
+                                                .setImage(dockerConfig.getImage())
+                                                .setNetwork(Protos.ContainerInfo.DockerInfo.Network.valueOf(dockerConfig.getNetwork()))
+                                )
                 )
-                .setCommand(command())
+                .setCommand(commandInfoMesosProtoFactory.create(resources))
                 .build();
     }
 
-    private Protos.CommandInfo command() {
-        return commandInfoMesosProtoFactory.create()
-                .setContainer(Protos.CommandInfo.ContainerInfo.newBuilder().setImage(dockerImage))
-                .build();
-    }
+    private Iterable<? extends Protos.Parameter> addParameters(Protos.Environment environment) {
+        // Do any of the parameters contain env vars? We need to inject them, because the docker command runs before we're in the sandbox
+        Map<String, String> dockerParameters = injectEnvVarsIntoParameters(environment);
 
-    private Iterable<? extends Protos.ContainerInfo.DockerInfo.PortMapping> portMappings(List<Protos.Resource> resources) {
-        Iterator<String> portsIterator = mesosConfig.getResources().getPort().iterator();
-        return resources.stream()
-                .filter(Protos.Resource::hasRanges)
-                .filter(resource -> resource.getName().equals("ports"))
-                .flatMap(resource -> resource.getRanges().getRangeList().stream())
-                .flatMapToLong(range -> LongStream.rangeClosed(range.getBegin(), range.getEnd()))
-                .limit(mesosConfig.getResources().getPort().size())
-                .mapToObj(hostPort -> Protos.ContainerInfo.DockerInfo.PortMapping.newBuilder().setHostPort((int) hostPort).setContainerPort(Integer.parseInt(portsIterator.next())).build())
-                .peek(portMapping -> logger.debug("Mapped host=" + portMapping.getHostPort() + "=>" + portMapping.getContainerPort()))
+        return dockerParameters.entrySet().stream()
```

Comment:
I can't see what we need this for. Only example given is to expose ports in Bridge mode, but that should already have been done in the Docker Image